src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -407,6 +408,16 @@ static bool can_crop_be_optimized_along_batch(const crop_node& node) {
     return false;
 }
 
+static bool can_read_value_be_optimize(const read_value_node& node) {
+    if (node.get_users().size() == 1)
+        return true;
+
+    if (node.get_users().back()->is_type<shape_of>())
+        return true;
```

Comment:
Why do you check type of the last user only? Isn't this optimization supposed to check that ReadlValue has all users of ShapeOf type OR all users of ShapeOf type except one?

Reply:
Yes, agree. 

something like this:
`int non_shape_of_users_count = xx; 
if (non_shape_of_users_count <= 1) return true;`

Reply:
update it as suggested